modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java

```diff
@@ -494,6 +566,63 @@ public void groupKey(int grpId, byte[] encGrpKey) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public IgniteFuture<Void> changeMasterKey(String masterKeyName) {
+        if (ctx.clientNode())
+            throw new UnsupportedOperationException("Client and daemon nodes can not perform this operation.");
+
+        if (!ctx.state().clusterState().active())
+            throw new IgniteException("Master key change was rejected. The cluster is inactive.");
+
+        checkMasterKeyChangeSupported();
+
+        if (masterKeyName.equals(getMasterKeyName()))
```

Comment:
I think we should complete the future with the error here.
Change master key is the maintenance procedure that executes on timeout of key compromise.
If the change was not done user should know about it.

Reply:
Done.